lib/src/usage_impl.dart

```diff
@@ -283,8 +348,24 @@ abstract class PersistentProperties {
 /// The `Future` from [sendPost] should complete when the operation is finished,
 /// but failures to send the information should be silent.
 abstract class PostHandler {
-  Future sendPost(String url, Map<String, dynamic> parameters);
+  Future sendPost(String url, List<Map<String, dynamic>> batch);
 
   /// Free any used resources.
   void close();
 }
+
+class _BatchingQueue {
+  List<Map<String, String>> enqueuedEvents = [];
+  int maxEventsPerBatch;
+
+  _BatchingQueue({required this.maxEventsPerBatch});
+
+  bool get isFull => enqueuedEvents.length >= maxEventsPerBatch;
```

Comment:
Maybe a todo here about item size, there is a max size that is relevant here as well.

If we send errors, then this max size might very well become relevant.

Reply:
From: https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide#batch-limitations

>* A maximum of 20 hits can be specified per request.
>* The total size of all hit payloads cannot be greater than 16K bytes.
>* No single hit payload can be greater than 8K bytes

Reply:
Done

Reply:
I implemented the 16k limit. Cannot find docs for the single-hit limit for single hits - so I'm not sure if there is really much we can do about those.